### PR TITLE
V2 final polish

### PR DESCRIPTION
**Pull request by drichelson**
See https://catamorphic.quip.com/Ggk6AbylpJBf for details. Passes v2 integration harness locally (with version comparison)

### CODE REVIEW

File: src/main/java/com/launchdarkly/client/FeatureFlag.java [hunk 1]
```diff
@@ -51,9 +53,6 @@ static FeatureFlag fromJson(String json) {
   }
 
   EvalResult evaluate(LDUser user, FeatureStore featureStore) throws EvaluationException {
-    if (user == null || user.getKey() == null) {
```
Comment (drichelson) [inline src/main/java/com/launchdarkly/client/FeatureFlag.java]: I pushed this up into the client code so it happens first thing. This is more consistent with the go client behavior and avoids a call to the feature store.

File: src/main/java/com/launchdarkly/client/StreamProcessor.java [hunk 2]
```diff
@@ -56,43 +56,47 @@ public void onOpen() throws Exception {
...
-          store.upsert(data.key(), data.feature());
-        }
-        else if (name.equals(DELETE)) {
-          FeatureDeleteData data = gson.fromJson(event.getData(), FeatureDeleteData.class);
-          store.delete(data.key(), data.version());
-        }
-        else if (name.equals(INDIRECT_PUT)) {
-          try {
-            store.init(requestor.makeAllRequest());
+        switch (name) {
```
Comment (drichelson) [inline src/main/java/com/launchdarkly/client/StreamProcessor.java]: converted to more idiomatic switch statement

File: src/main/java/com/launchdarkly/client/FeatureRequestEvent.java [hunk 3]
```diff
@@ -8,9 +8,17 @@
   @SerializedName("default")
   JsonElement defaultVal;
 
-  FeatureRequestEvent(String key, LDUser user, JsonElement value, JsonElement defaultVal) {
+  @SerializedName("version")
```
Comment (jkodumal) [inline src/main/java/com/launchdarkly/client/FeatureRequestEvent.java]: Extra line?

**Comment by jkodumal** [comment]
👍

**Pull request merged**